logic/transaction_pool.js

```diff
@@ -703,27 +713,32 @@ TransactionPool.prototype.undoUnconfirmedList = function(cb, tx) {
  * @returns {SetImmediate} error, ids[]
  */
 TransactionPool.prototype.expireTransactions = function(cb) {
-	async.waterfall(
-		[
-			function(seriesCb) {
-				__private.expireTransactions(
-					self.getUnconfirmedTransactionList(true),
-					seriesCb
-				);
-			},
-			function(seriesCb) {
-				__private.expireTransactions(
-					self.getQueuedTransactionList(true),
-					seriesCb
-				);
-			},
-			function(seriesCb) {
-				__private.expireTransactions(
-					self.getMultisignatureTransactionList(true),
-					seriesCb
-				);
-			},
-		],
+	library.balancesSequence.add(
+		balancesSequenceCb => {
+			async.waterfall(
+				[
+					function(waterfallCb) {
+						__private.expireAndUndoUnconfirmedTransactions(
+							self.getUnconfirmedTransactionList(true),
+							waterfallCb
+						);
+					},
+					function(waterfallCb) {
+						__private.expireTransactions(
+							self.getQueuedTransactionList(true),
+							waterfallCb
+						);
+					},
+					function(waterfallCb) {
+						__private.expireTransactions(
+							self.getMultisignatureTransactionList(true),
+							waterfallCb
+						);
+					},
+				],
+				balancesSequenceCb
+			);
+		},
 		() => setImmediate(cb)
```

Comment:
No error handling here. Even if you think there could't be any error, but still handle the errors appropriately.